### DISCUSSION TITLE
jekyll and theme only apply to /README.md, not to other pages

### DISCUSSION THREAD

**Discussion by jneea-new** [community/community#125293] [Programming Help]
### Select Topic Area

Question

### Body

Hello,

I have some trouble to build a site on GitHub pages. I fear that stuff wasn't correctly initialized (for jekyll), maybe because I activated the service 2 years ago but start only now to fill in some content. When I activated the service, only an `index.html` file was created, no `README.md` and especially no config files. It seems (from what I read in the documentation and "quickstart tutorials") that today, the activation process also creates the `_config.yml` file, maybe others?

So I deleted the `index.html` file and created a `README.md` file instead.

I also created a `_config.yml` file in which I define the name, theme & description of my website. (In the tutorials it seems that this config file is filled with lots of stuff, but since I couldn't fine one, I made my own with only these three lines.)

Now that seems to work halfways : the theme is indeed applied to the `README.md` file, although it has no YAML header(!):
When I visit the site, I see its (the `README.md` file's) contents, well formatted by Jekyll (I guess) and wrapped inside page header & footer & navigation sidebar , according to the chosen theme.

But when I create any other file, say `about.md` (actually : should I make a file `about.md` or rather a folder `/about/` and a `README.md` file in that folder? would that have the same effect....?) -- but that file is ignored by jekyll. When I visit my site and go to url `/about`, the MD markup is processed (specifically, `[xxx](yyy)` is turned into links), but it is displayed as black text inside an entirely white window/background, with none of the theme's formatting (header, footer, sidebars ...) .

The YAML header (where I put `title:"About"`; `layout: page`... between `---`) is completely ignored and stripped when I view the page. (So there isn't even a title.)

What am I doing wrong? Should I "unpublish" the site and re-activate pages in the hope that stuff will get initialized correctly (if that is the problem)?

(Also, I'm confused, since part of the documentation speaks about .html pages I should create, and another part speaks of .md pages.  Part of the documentation speaks about installing jekyll on my own computer (which I did), the other part seems to suggest that I can do everything "online" on gitHub if I don't want additional benefit from editing and testing on my own computer before committing to github.

I'm as confused that jekyll does process the `README.md` file correctly (although I did not put the YAML header with title, layout info), as I am confused that it does NOT process the `about.md` file where I did put that YAML header.

Thanks a lot in advance for your help and sorry for the noob question.

**Comment by LiteBrite82** [comment]
Thanks for posting in the GitHub Community, @jneea-new ! 

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.
